utils/create_ellipsoid_dataset.py

```diff
@@ -0,0 +1,230 @@
+import os
+import json
+import argparse
+
+from tqdm import tqdm
+import numpy as np
+import itk
+import vtk
+from itkwidgets import view
+from vtk.util import numpy_support
+
+
+def center_and_align(surf_pts, srep_pts):
+    # Center surface and srep at the origin
+    surf_center = np.mean(surf_pts, axis=0)
+    surf_pts_centered = surf_pts - surf_center
+
+    srep_pts_centered = srep_pts - surf_center
+
+    # Axis align the templates
+    u, s, v = np.linalg.svd(surf_pts_centered.transpose() @ surf_pts_centered)
+    R = v.transpose() @ u.transpose()
+
+    surf_pts_align = np.zeros(surf_pts.shape)
+    for i in range(surf_pts.shape[0]):
+        surf_pts_align[i, :] = u.transpose() @ surf_pts_centered[i, :]
+
+    srep_pts_align = np.zeros(srep_pts.shape)
+    for i in range(srep_pts.shape[0]):
+        srep_pts_align[i, :] = u.transpose() @ srep_pts_centered[i, :]
+
+    return surf_pts_align, srep_pts_align
+
+
+def scale_along_axes(surf_pts, srep_pts, scale_std=0.15):
+    # Apply scaling
+    # Expects the surf_pts and srep_pts to be aligned and centered.
+    axis_scales = np.abs(np.random.normal(1.0, scale_std, size=3))
+    xscale, yscale, zscale = axis_scales[0], axis_scales[1], axis_scales[2]
+    surf_pts_scaled = np.zeros(surf_pts.shape)
+    surf_pts_scaled[:, 0] = surf_pts[:, 0]*xscale
+    surf_pts_scaled[:, 1] = surf_pts[:, 1]*yscale
+    surf_pts_scaled[:, 2] = surf_pts[:, 2]*zscale
+
+    srep_pts_scaled = np.zeros(srep_pts.shape)
+    srep_pts_scaled[:, 0] = srep_pts[:, 0]*xscale
+    srep_pts_scaled[:, 1] = srep_pts[:, 1]*yscale
+    srep_pts_scaled[:, 2] = srep_pts[:, 2]*zscale
+
+    axis_scales = (xscale, yscale, zscale)
+    return surf_pts_scaled, srep_pts_scaled, axis_scales
+
+
+def bend_and_twist(surf_pts_scaled, srep_pts_scaled, std_th=np.pi/12, std_ph=np.pi/8):
```

Comment:
It is not obvious to me what `th` and `ph` are, although it looks like `std` means standard deviation. It would be nice to have more descriptive names or at least a comment saying what they mean. If the names are brutally obvious/very standard to someone with more domain experience than me, then I guess go the comment route.